### PR TITLE
Server: Move EULA specific code into its own file

### PR DESCRIPTION
**Pull request by fluvf**
I tried to not break anything in the web interface, as it reads values straight from the ´MinecraftServer´ instance. And from what I tested, nothing should be broken.
Nothing visible anyways :D

I'd like to get your opinion on some changes.
- What do you think about the API? Should the paths be changed and new ones added for the `Eula` instance?
If / when support for multiple maps / servers is added, those will have to change anyways, so waiting till then could save us from some extra work.
- I think moving towards async code would be beneficial.
If `async/await` is used, it shouldn't make the code any harder to understand and, if done properly and with good documentation, it would result in cleaner and faster code.
- I know this is a bit early and this isn't the right place for it, but I thought about it when making this, so I'll write it down.
How would you feel about moving the whole web interface into its own repo? It already kind of is it's own package. Maybe something like `npm run install [path]` could be added, that builds and moves the interface into a folder, where it could then be used with the server and API. That would also force writing api based code, so it doesn't rely on internal variables / functionality.

### CODE REVIEW

**Comment by nickrnet** [comment]
1. If we want to break out classes in the API, we should probably investigate ExpressJS routing and separate things from the API vs. Minecraft state. I have been meaning to go down this rabbit hole anyway once work on the Preferences tab of the web app starts progressing.
2. I fully agree with the async changes. We will have to tweak both the `minecraft-server `and `minecraft-api` and fully regression test, but no bigs.
3. Is the intent to have the api and server classes be `npm install`-able? I haven't given that much thought but could be useful and would guide the build process quite a bit. There's work to do there to let the api and server be independent of the web app and vice versa, but doable.

**Comment by nickrnet** [comment]
Also if we are going to have classes of `minecraft-server.properties`, we should invent a directory structure, something like
```
<root>/
├ doc/
├ dist/                                                # holds the built parts
├ src/
├── api/                                          # holds minecraft-api.js parts
├── minecraft/                               # holds minecraft-server.js parts
├── minecraft-server-manager/       # holds the webapp parts
├── minecraft-server/                       # holds the minecraft installation
├ ...
```
Thoughts?

**Comment by nickrnet** [comment]
I added a "WIP" to the front of this PR so I don't merge it until you remove it.

**Comment by fluvf** [comment]
(I'm not too eloquent with english, so I'm sorry if the bellow seems a bit too harsh or hard to follow)

As far as the folder structure goes, I feel something like this would be good:
> docs \\
> // Holds all scripts and the data they generate, as not to create conflicts with the Minecraft Server executable
> manager \\
>   // Holds unloaded maps (preferably as compressed files)
>   -\ maps \\
>   // Holds .png files that would be used as the server icon seen in Minecrafts mutiplayer tab
>   -\ icons \\
>   // Holds Minecraft Server .jar executables
>   -\ servers \\
>   // Holds backups of all maps. Filenames should consist of a timestamp and the world name
>   -\ backups \\
>   // Holds automatically managed log files (preferably as compressed files)
>   -\ logs \\
> // Holds all files generated by the Minecraft Server executable
> server \\
>   -\ server_name_foo \\
>     ...

At some point it would make sense to have the manager logic be npm installable. That has also influenced the folder structure above.
- I don't see a reason why the server side files would need compiling, combining or some other build process before they'd be ready for use. If they're to be npm installable, I feel like the folder structure should stay as simple and as "shallow" as possible.
- I don't really know what we would gain from separating the API into multiple files. My idea was to move most logic that directly deals with the Minecraft server and its files into their own classes and have `minecraft-server.js` pretty much only manage configs. The api would then only allow calling all functions that `minecraft-server.js` has through a web interface
- If or when the web interface were to be separated from this project, I don't think it should need to know anything about the managers internal state. Just where to access the api and that's it. All the rest needed would come from api calls.

I would like this to be able to handle multiple maps and servers at the same time. To achieve that, the logic should most likely be map based. I know how it would work and I'm now trying to articulate it in #28, but it's a lot do :D
In short, to allow high level of configurability, the manager (`minecraft-server.js` atm) would load `.json` files that hold string id'd info about maps, servers, their versions, etc. When the user tries to load a map through the web interface, the manager would then go through those configs and move / create the files at the right folders.

**Comment by fluvf** [comment]
Finally not a WIP branch anymore! I added tests for the class and changed `minecraft-server.js` in a way the API can be left untouched

All of the commits are self contained, even if there's a couple of extra colons and variables in a few of them.
